docs/conf.py

```diff
@@ -14,11 +14,27 @@
 # -- General configuration ---------------------------------------------------
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
 
-extensions = ["sphinx.ext.autodoc", "sphinx.ext.viewcode", "sphinx.ext.napoleon"]
+extensions = [
+    "sphinx.ext.autodoc",
+    "sphinx.ext.viewcode",
+    "sphinx.ext.napoleon",
+    "sphinx.ext.intersphinx",
+    "sphinx_autodoc_typehints",
+]
 
 templates_path = ["_templates"]
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 
+napoleon_preprocess_types = True
+python_use_unqualified_type_names = True
+
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3/", None),
+    "numpy": ("https://numpy.org/doc/stable/", None),
+    "matplotlib": ("https://matplotlib.org/stable/", None),
+    "dolfinx": ("https://docs.fenicsproject.org/dolfinx/main/python", None),
+    "ufl": ("https://fenics.readthedocs.io/projects/ufl/en/stable/", None),
+}
 
 # -- Options for HTML output -------------------------------------------------
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
```

Comment:
[I can't comment on line 42 here because it's too far from one of your changes]

Am I allowed an aesthetics bike-shedding comment? 
How do you feel about the [PyData Sphinx Theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/index.html) ?

https://github.com/SainsburyWellcomeCentre/WAZP/blob/main/docs/source/conf.py#L88

Reply:
Would be happy to change to this; I mainly when with the readthedocs theme because it was familiar and I didn't particularly like the default Alabaster theme.

Reply:
Have switched to the PyData theme as I think it is a bit more readable, particularly on wider screens.